### PR TITLE
border: fix accessing union bug

### PR DESCRIPTION
**Pull request by ampresent**
Normally, field accessing like `task_struct.sched_entity.load` is correctly
handled in an AST tree like,

```plain
     ComponentRef
target |  field  \
       |          \
ComponentRef     load
       |    \
target |field\
       |      \
task_struct  sched_entity
```

However, when accessing field through a unnamed union, for example
`sched_class[.unamed_union].pick_next_task`. This is introduced in
RHEL/CentOS with the RH_KABI_REPLACE, which creates unions wrapper
for fields. Developers omit `[.unamed_union]` in the code because
they're unamed. We get an AST tree like,

```plain
     ComponentRef
target |  field  \
       |          \
ComponentRef     pick_next_task
       |    \
target |field\
       |      \
sched_class (unamed KABI replacing union)
```

Before this patch, collect.py finds
- ComponentRef1(sched_class, [unamed_union])
- ComponentRef2(CompnentRef1, pick_next_task)

This patch compress these two operations into one,
- ComponentRef(sched_class, pick_next_task)

Signed-off-by: Yihao Wu <wuyihao@linux.alibaba.com>

### CODE REVIEW

File: boundary/collect.py [hunk 1]
```diff
@@ -233,28 +233,34 @@ def mark_callback(op, caller):
...
         for var in gcc.get_variables():
-            decl = var.decl
-            type_name = '' if not decl.type.name else decl.type.name.name
-
-            # struct sched_class is purely private
-            if decl.initial and type_name != 'sched_class' and \
-                    not self.decl_in_section(decl, '.discard.addressable'):
-                decl.initial.walk_tree(mark_callback, decl)
+            if var.decl.initial and not self.decl_in_section(var.decl, '.discard.addressable'):
+                var.decl.initial.walk_tree(mark_callback, var.decl)
```
Comment (ampresent) [inline boundary/collect.py:237]: This was left by accident, I'll remove it

File: boundary/collect.py [hunk 2]
```diff
@@ -243,18 +243,29 @@ def mark_callback(op, caller):
 
     def collect_struct(self):
         public_fields = defaultdict(set)
+        parent_component_ref = {}
```
Comment (chenshanpei) [inline boundary/collect.py:246]: the scope of parent_component_ref is stmt.walk_tree, why it appears here?

Comment (ampresent) [reply boundary/collect.py:246]: This line doesn't change its scope or any functionality. I add it here just to improve readability.

**Pull request merged**